### PR TITLE
Hw 20 1

### PR DESCRIPTION
**Pull request by semyonich**

### CODE REVIEW

**Review by MykolaFatkullin — APPROVED** [review]

File: src/main/java/core/basesyntax/service/CarServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        return carDao.delete(id);
+    }
+
+    @Override
+    public void addDriverToCar(Driver driver, Car car) {
+        carDao.update(car).getDrivers().add(driver);
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/service/CarServiceImpl.java]: we should first modify the car and then call method update, not vice versa

File: src/main/java/core/basesyntax/service/CarServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    }
+
+    @Override
+    public void removeDriverFromCar(Driver driver, Car car) {
+        carDao.update(car).getDrivers().remove(driver);
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/service/CarServiceImpl.java]: same here

File: src/main/java/core/basesyntax/service/CarServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        List<Car> driverCars = new ArrayList<>();
+        for (Car car : carDao.getAll()) {
+            for (Driver driver : car.getDrivers()) {
+                if (driver.getId().equals(driverId)) {
+                    driverCars.add(car);
+                }
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/service/CarServiceImpl.java]: move this logic to dao layer, we should receive filtered data from dao

**Review by RomanDubovskyi — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/service/CarServiceImpl.java (hunk 1 above)
Comment (semyonich) [reply src/main/java/core/basesyntax/service/CarServiceImpl.java]: done

File: src/main/java/core/basesyntax/service/CarServiceImpl.java (hunk 2 above)
Comment (semyonich) [reply src/main/java/core/basesyntax/service/CarServiceImpl.java]: done

File: src/main/java/core/basesyntax/service/CarServiceImpl.java (hunk 3 above)
Comment (semyonich) [reply src/main/java/core/basesyntax/service/CarServiceImpl.java]: done

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**